packages/app/src/app/pages/Dashboard/Content/index.tsx

```diff
@@ -57,9 +57,15 @@ export const Content = withRouter(({ history }) => {
       })}
     >
       <Switch>
-        {isFeatureFlagBeta && (
-          <Route path="/dashboard/beta" component={BetaRepositoriesPage} />
-        )}
+        <Route
+          path="/dashboard/beta"
+          /**
+           * Wrap the component instead of the whole route, as the
+           * feature-flag query takes a few ms to load, and meanwhile the Route
+           * redirect the page because the route hasn't been registered yet
+           */
+          component={isFeatureFlagBeta ? BetaRepositoriesPage : () => null}
```

Comment:
Would something like this work to avoid the possibility of rendering a blank page?

```suggestion
          component={isFeatureFlagBeta ? BetaRepositoriesPage : () => <Redirect to={dashboardUrls.home(activeTeam)} />}
```

Reply:
This would end up with the same behavior as before because the Redirect would be rendered anyway. A final and right solution would need some kind of loading state, but I think it's over-engineered for this beta version in a scenario that we control who has access to this.